WalletWasabi.Fluent/ViewModels/CoinJoinProfiles/CoinJoinProfilesViewModel.cs

```diff
@@ -14,23 +15,44 @@ namespace WalletWasabi.Fluent.ViewModels.CoinJoinProfiles;
 [NavigationMetaData(Title = "Coinjoin Strategy")]
 public partial class CoinJoinProfilesViewModel : RoutableViewModel
 {
+	private readonly KeyManager _keyManager;
+
 	[AutoNotify] private CoinJoinProfileViewModelBase? _selectedProfile;
+	[AutoNotify] private string _plebStopThreshold;
 
-	public CoinJoinProfilesViewModel(KeyManager keyManager)
+	public CoinJoinProfilesViewModel(KeyManager keyManager, bool selectDefaultProfile)
 	{
-		NextCommand = ReactiveCommand.Create(() => OnNext(keyManager));
-		EnableBack = true;
+		_keyManager = keyManager;
+		_plebStopThreshold = (keyManager.PlebStopThreshold?.ToString() ?? KeyManager.DefaultPlebStopThreshold.ToString()).TrimEnd('0');
 
-		var speedyProfile = new SpeedyCoinJoinProfileViewModel();
+		NextCommand = ReactiveCommand.Create(OnNext);
+		EnableBack = true;
 
 		Profiles = new()
 		{
 			new EconomicCoinJoinProfileViewModel(),
-			speedyProfile,
+			new SpeedyCoinJoinProfileViewModel(),
 			new PrivateCoinJoinProfileViewModel()
 		};
 
-		_selectedProfile = speedyProfile;
+		_selectedProfile =
+			Profiles.Where(x => x.MinAnonScoreTarget == keyManager.MinAnonScoreTarget)
+					.Where(x => x.MaxAnonScoreTarget == keyManager.MaxAnonScoreTarget)
+					.Where(x => x.FeeRateMedianTimeFrameHours == keyManager.FeeRateMedianTimeFrameHours)
+					.FirstOrDefault();
+
+		if (_selectedProfile == null && selectDefaultProfile)
+		{
+			_selectedProfile = Profiles[1];
+		}
+
+		if (_selectedProfile == null)
+		{
+			SelectedManualProfile = new ManualCoinJoinProfileViewModel(keyManager.AutoCoinJoin, keyManager.MinAnonScoreTarget, keyManager.MaxAnonScoreTarget, keyManager.FeeRateMedianTimeFrameHours);
+		}
+
+		this.WhenAnyValue(x => x.SelectedProfile, x => x.SelectedManualProfile)
+			.Subscribe(_ => ApplyChanges());
```

Comment:
- It is unnecessary to set a manual coinjoin profile because if the user changes the selection it will be automatically created.
- Skip added to avoid unnecessary file save, because every time when the dialog was opened, it saved the wallet immediately, it shouldn't happen, mainly when we are in the wallet creation process. If the wallet got saved but the user presses the cancel, next when they open the software a nonwanted wallet will appear!

```suggestion
		var defaultProfile = Profiles[1];
		var currentProfile =
			Profiles.Where(x => x.MinAnonScoreTarget == keyManager.MinAnonScoreTarget)
				.Where(x => x.MaxAnonScoreTarget == keyManager.MaxAnonScoreTarget)
				.Where(x => x.FeeRateMedianTimeFrameHours == keyManager.FeeRateMedianTimeFrameHours)
				.FirstOrDefault()
			?? defaultProfile;

		_selectedProfile = selectDefaultProfile ? defaultProfile : currentProfile;

		this.WhenAnyValue(x => x.SelectedProfile, x => x.SelectedManualProfile)
			.Skip(1)
			.Subscribe(_ => ApplyChanges());
```

Reply:
@soosr 

> It is unnecessary to set a manual coinjoin profile because if the user changes the selection it will be automatically created.

This is done to show the current values in the dialog, otherwise you get bad UX because it seems to the user as if values were not really saved. For instance:

 - Set MinAnonScoreTarget to 20
 - click Done
 - close settings
 - re-open settings
 - you don't see the 20

Reply:
You are absolutely right, I forgot about when they use it from the wallet settings.